### DISCUSSION TITLE
A question on the point sampling strategy

### DISCUSSION THREAD

**Discussion by Totoro97** [NVlabs/instant-ngp#958] [Q&A] [answered]
Hi thanks for this excellent project! It is really impressive.

I have a dumb question on the point sampling strategy. In the `fox` dataset, I set the `aabb_scale` as 128, which is a really large value.
And as shown in the visualization the occupancy grids are very dense (the image below). However, the sampled points number per ray is amazingly low (~80 points/ray). 

![image](https://user-images.githubusercontent.com/6695053/191504260-601ebcc4-7d41-457e-9874-52e49822371c.png)

For reference I also see that the samples per ray where `aabb_scale=4` is ~70.
![image](https://user-images.githubusercontent.com/6695053/191505632-0fa7f860-28e7-4347-a265-66276da0228a.png)

What confused me is that dramatically increasing the `aabb_scale` does not really affect the number of point samples. Is there any other point sampling strategy besides exponential stepping and occupancy grid skipping?

Thanks in advance!

**Comment by Tom94** [comment, marked as answer]
Hi there, yes, there is an additional termination of rays once they become opaque (i.e. transmittance gets close to zero). Cheers!

**Reply by Totoro97** [reply]
Thanks!
